database/sql/V048__create_settlement_rewards.sql

```diff
@@ -0,0 +1,29 @@
+-- Populated before settlement occurs on chain
+-- (after the submissions have been ranked)
+CREATE TABLE settlement_scores (
+  settlement_id bytea NOT NULL,
+  winning_score numeric(78,0) NOT NULL,
+  reference_score numeric(78,0) NOT NULL
+);
+
+-- Populated after block finalization via transactionReceipt.
+CREATE TABLE settlement_observations (
+  -- the appended settlement ID read from the call data
+  settlement_id bytea NOT NULL,
```

Comment:
Is this settlement_id the name we want for this long term? In `crates/driver/openapi.yml` this is called `transaction_identifier`.

Reply:
Sounds fine with me, as long as it's "id" and not "identifier" - since id means the same thing and is way shorter.

Reply:
I'm actually more worried that the EVM can not enforce this id to exist and so it might not always be there. Perhaps we should consider allowing it to be nullable. We can identify the transaction and solver by the event --- although I guess then there are other fields that could not be linked (the reference score). 

If we were to follow traditional financial accounting principles we would have to make these two fields nullable and someone would have to manually try to match missing records at the end of each accounting period as best as possible.

Reply:
Technically, instead of being null it would be an empty byte array. So either way it doesn't need to be nullable but in some situations there might not be a meaningful id.

Reply:
Another discussion is whether this should be UNIQUE. The way we used to talk about the transaction_identifier, is that we needed it as a way to identify settlement transactions when solvers settle them. This means that from the perspective of the backend they only have to be unique temporarily but it is not a problem if we reuse these later.

Reply:
> Sounds fine with me with a strong preference for "id" and not "identifier" - since it means the same thing and is way shorter.

id or identifier is fine. But first word word should match the rest of the code.

Reply:
> Another discussion is whether this should be UNIQUE.

If we read that value from the block chain we can't guarantee that it's unique. Some external party might have a bug and reuse an old id or there is an adversary trying to mess with us by submitting settlements with duplicated ids on purpose.

Reply:
While it it true that is may not be unique -- we are going to have a pretty hard time (non-deterministic) matching things which are not. We may want to come up with some preventative measures or at least an approach for handling these kinds of situations early on.

Reply:
Note that I am changing both occurrences of `settlement_id` to `transaction_id` now. I hope this naming convention is really what we want.

```
CREATE TABLE settlement_scores (
  transaction_id bytea NOT NULL,
  winning_score numeric(78,0) NOT NULL,
  reference_score numeric(78,0) NOT NULL
);

-- Populated after block finalization via transactionReceipt.
CREATE TABLE settlement_observations (
  -- the appended transaction identifier read from the call data
  transaction_id bytea NOT NULL,
```

Reply:
Changed name here, but there seems to be a few other things in this thread -- feel free to "resolve" whenever anyone feels its all been sorted.